android/src/main/java/io/ably/lib/push/LocalDevice.java

```diff
@@ -15,9 +15,10 @@
 import io.ably.lib.types.RegistrationToken;
 import io.ably.lib.util.Base64Coder;
 import io.ably.lib.util.Log;
+import io.ably.lib.util.Serialisation;
 
 public class LocalDevice extends DeviceDetails {
-    public String deviceSecret;
```

Comment:
This field is moved to DeviceDetails as protocol 2.0 requires it now.

Reply:
Yes, but the new protocol doesn't change the IDL for this. I recommend you use `LocalDevice` instead of `DeviceDetails` in transport. That's what I did in ably-cocoa

Reply:
LocalDevice is only part of the Android build so it is not present in core tests. That is why we cannot just use LocalDevice. One solution was to make LocalDeviceBase class in core and override it in Android and Java builds (java not needed) or to just move one field in parent class and implement it in LocalDevice which is android only.

Reply:
I personally do not like that a `LocalDevice` that extends a class called `DeviceDetails`, it would be preferrable to have an interface that represents LocalDevice but I suppose we have to follow specs. 

> LocalDevice is only part of the Android build so it is not present in core tests

I wonder what sort of tests exist to test DeviceDetails 🤔 Isn't it supposed to be used for push notifications only.

>  That is why we cannot just use LocalDevice. One solution was to make LocalDeviceBase class in core and override it in Android and Java builds

Instead of doing this, why don't you move `LocalDevice` to core library, provided that it has relevant tests existing in core class, it should be part of it too.

Reply:
Not sure if it is ok to move it to core. @KacperKluka will know better what to do here.

Reply:
According to the [spec about LocalDevice](https://sdk.ably.com/builds/ably/specification/main/features/#RSH8):
> In platforms that **support receiving push notifications**, the `device` method on the `RestClient` or `RealtimeClient` interfaces returns an instance of `LocalDevice` that represents the current state of the device in respect of it being a target for push notifications.

Due to this, I'm not sure if the `LocalDevice` should be present in both `ably-java` and `ably-android` 🤔 The other problem is that the `LocalDevice` uses Android-specific classes (such as `ActivationContext`) which should not be moved to the core library, as they are Android only 😉

Reply:
@KacperKluka I have opened the issue about deviceSecret and you can read more here https://github.com/ably/ably-java/issues/845

Reply:
Ok. I also created an issue on the documentation repo. https://github.com/ably/specification/issues/110. I'll resolve this conversation in order to unblock this PR.